pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -613,7 +613,16 @@ int kretprobe__sockfd_lookup_light(struct pt_regs* ctx) {
     struct socket* socket = (struct socket*)PT_REGS_RC(ctx);
     enum sock_type sock_type = 0;
     bpf_probe_read(&sock_type, sizeof(short), &socket->type);
-    if (sock_type != SOCK_STREAM) {
+
+    struct proto_ops *proto_ops = NULL;
+    bpf_probe_read(&proto_ops, sizeof(proto_ops), &socket->ops);
```

Comment:
I don't think this read has a stable offset, because it is preceded by a `struct file *`. How important is the family check?

Reply:
I think it's fair to be concerned about the stability of `struct file`, but not a _pointer_, right?

Reply:
> but not a _pointer_, right?

Ah, true. Looking at 4.4, there is an additional struct member, so it isn't stable anyways: https://elixir.bootlin.com/linux/v4.4/source/include/linux/net.h#L119

Reply:
We might be able to build an offset guess for the `struct sock *` field, and thus the `const struct proto_ops *` field right after it.

Reply:
Argh. Good catch

Reply:
Added fix using existing offset for `(struct socket).sk`, plus `sizeof(void*)`.